docs/_docs/reference/changed-features/implicit-resolution.md

```diff
@@ -165,7 +165,22 @@ Condition (*) is new. It is necessary to ensure that the defined relation is tra
 
 [//]: # todo: expand with precise rules
 
-**9.** The following change is currently enabled in `-source future`:
+
+**9.** Given disambiguation has changed. When comparing two givens that both match an expected type, we used to pick the most specific one, in alignment with
+overloading resolution. From Scala 3.5 on, we pick the most general one instead. Compiling with Scala 3.5-migration will print a warning in all cases where the preference has changed. Example:
```

Comment:
This is a separate PR: https://github.com/scala/scala3/pull/19300

It is rolled into this PR because this PR contains examples where we have triangles such as in the parser combinators, where two type classes extend Combinator. The new rules are needed to resolve these better. And they seem not to break anything, empirically! https://github.com/scala/scala3/pull/19300 contains a discussion in the comments.